### ISSUE TITLE
Redirect '/login' to frontpage if user logged-in

### ISSUE THREAD

**Issue by jeff1evesque** [jeff1evesque/machine-learning#2860]
This issue is a partial continuation of #2818.  Specifically, if the user has already logged-in, then an attempt to access the `/login` page, should result in a redirect to the applications designated frontpage.

**Comment by jeff1evesque** [comment]
Since we have implemented react-router, we need to consider integrating the `browserHistory.push` to redirect users to the frontpage, if the redux store contains a key indicating the user has already logged-in.

**Comment by jeff1evesque** [comment]
cced588: upon `/login`, we receive the following console error:

>Uncaught (in promise) TypeError: Cannot read property 'push' of undefined

![login-attempt](https://cloud.githubusercontent.com/assets/2907085/22183226/dbd08f1a-e086-11e6-9830-53ebbfe3cf40.PNG)

**Comment by jeff1evesque** [comment]
Upon immediate `/login`, the user is redirected to the defined homepage.  However, after being logged-in, visiting the `/login` page will not redirect the user to the homepage.

**Comment by jeff1evesque** [comment]
We've got our redirect working.  However, if we `/login` initially, we get the following error:

![redirect](https://cloud.githubusercontent.com/assets/2907085/22185122/7f080eda-e0ad-11e6-99be-2f9f6d2878dc.PNG)

**Note:** since we don't have a dedicated frontpage component, the login component still is rendered - though, if we pay close attention, the url in the browser has adjusted correctly.

After, if we visit the `/login` url, we will be redirected to the frontpage url `/` without the above errors.

**Comment by jeff1evesque** [comment]
If we were at the main frontpage, clicked the login button, and successfully `/login`, we are redirected to the corresponding frontpage:

![login](https://cloud.githubusercontent.com/assets/2907085/22185679/f0a6448a-e0b7-11e6-9b4b-c8e8ce2056b7.PNG)

**Note:** if we were on the `/register` page, then click the login button, followed by a successful `/login`, we would be redirected back to the `/register` page.  This is partially because we need to define an exclusive frontpage component, and implement its reference with the `router.jsx`.

Additionally, we made a temporary alteration to `login.jsx`, by ensuring the existence of `browserHistory.push('/session/data-new')` within `componentWillMount`:

```javascript
...
    componentWillMount: function() {
      // load username from redux: user already logged-in
        if (
            this.props &&
            this.props.username != 'anonymous'
        ) {
            var username = this.props.username;
        }
      // load username from sessionStorage: maybe browser reloaded
        else if (
            this.props &&
            this.props.username == 'anonymous' &&
            loadState('username') &&
            String(loadState('username')) != 'anonymous'
        ) {
            var username = loadState('username')
            var action = setLoginState(username);
            this.props.dispatch(action);
        }

      // redirect to homepage if logged-in
        if (username && username != 'anonymous') {
            browserHistory.push('/session/data-new');
        }
    },
...
```

Now, if we attempt to access the `/login` page, while being logged-in, we are redirected as follows:

![logged-in](https://cloud.githubusercontent.com/assets/2907085/22185704/675f7ae2-e0b8-11e6-96cd-7eb01c538047.PNG)

So, we can revert the above `browserHistory.push('/session/data-new')` to `browserHistory.push('/')`.  Additionally, the user experience error we noticed, upon initial login, where specific reactjs components were missing, will be resolved as we make `render` logic contingent on our redux store.  This will be more consistent then relying on passing properties between parent, and children react components.

**Note:** the following issues will help further enhance the above initial login experience:

- #2857 
- #2851 
- #2885
